### ISSUE TITLE
Only allow top-level browsing contexts (navigable) in session subscribe/unsubscribe 

### ISSUE THREAD

**Issue by whimboo** [w3c/webdriver-bidi#108]
As of now the [session.subscribe](https://w3c.github.io/webdriver-bidi/#command-session-subscribe) command doesn't make it clear that only top-level BrowsingContexts (navigables) should be passed in for enabling event subscription for the whole BrowsingContext tree. Instead we [implicitly get these from a given BrowsingContext when updating the events map](https://w3c.github.io/webdriver-bidi/#update-the-event-map) (step 7.2.2).

It would be better to only allow top-level BrowsingContexts to be passed-in, and error out respectively if that isn't the case.

The same also applies to `session.unsubcribe`.

**Comment by foolip** [comment]
This seems similar to https://github.com/w3c/webdriver-bidi/issues/72, where one of the options were:
> Only allow subscribing to events per top-level browsing context, not for any context. That makes things simpler (although there is still a decision to make if you first subscribe for a specific context and then later globally) and more or less matches what CDP has (but CDP doesn't smooth over out of process iframes; I assume we'd want to do that).

Where *I* is @jgraham.

**Comment by foolip** [comment]
Aha. An error seems much preferable in this case, unless subscribing for a descendant browsing context is a handy shorthand in some common case we can foresee? And even then, it seems wise to reserve subscription for such browsing contexts in case we want to support it later.

**Comment by jgraham** [comment]
Yeah, I'm not opposed. The thinking behind the current scheme was that if you have the handle to a context and want events for that context, this will allow you to get them in one command whereas if it errors you need to first get the tlbc and then subscribe for that. I don't know in practive how likely it is to have the id for a context but not it's top level context (I suppose we could also make things that return contexts always return the handle of the parent and top level context, so it's impossible).

**Comment by Lightning00Blade** [comment]
@jgraham @sadym-chromium I think we should revisit this as the current algorithm if we have a non top-level context we subscribe on the top-level, that will mean we emit for each of it's children. 
https://w3c.github.io/webdriver-bidi/#update-the-event-map

1. Top Level Context A
   1. Child Frame B
   2. Child Frame C
2. Subscribe to C
3. Get events for A, B, C

I think this is an unexpected behavior from user point of view.
